lib/syskit/task_context.rb

```diff
@@ -959,6 +983,14 @@ def setup_failed!(exception)
 
             state_reader.resume if state_reader.respond_to?(:resume)
 
+            unless needs_setup?
+                promise = promise(description: "promise:#{self}#start")
+                commit_properties(promise)
+                promise.then { orocos_task.start(false) }
+                start_event.achieve_asynchronously(promise, emit_on_success: false)
```

Comment:
Well. In read_only mode, we should ... not start the component.

Just return is fine.

Reply:
> so the component will always be on ":PRE_OPERATIONAL".

No. It will be in RUNNING once the syskit that controls that component (a.k.a. has a non-readonly taskcontext on it) will start it.

In tests, you start the component manually within the test, just as I did in the code snippet I sent on the chat

Reply:
> In tests, you start the component manually within the test, just as I did in the code snippet I sent on the chat

okay `orocos_task.start(false) never emits`

```rb
Orocos.allow_blocking_calls do
                expect_execution { task.orocos_task.start(false) }
                    .to { emit task.start_event }
            end
```

The only way that I achieve to get the emission when using a "read_only" task was:

```rb
#Syskit::TaskContext
 event :start do |_context|
            info "starting #{self}"

            if read_only?
                event(:start).emit
                return
            end
            
[....]
```

 and then calling:

```
Orocos.allow_blocking_calls do
                expect_execution { task.start! }
                    .to { emit task.start_event }
end
```

But it's not really starting the component though...

Reply:
Unless I should not look into "task.orocos_task.rtt_state", but "task.started?" (from Roby::Task), is that so?

Reply:
An update:
Calling start(false) directly inside `allow_blocking_calls` does start the component

````rb
Orocos.allow_blocking_calls do
                task.orocos_task.start(false)
end
```

Reply:
Just to be sure, you do need to also start the task, i.e.

```
expect_execution do
  task.orocos_task.start(false)
  task.start!
end.to { emit task.start_event }
```

If the event is not emitted, you need to find out why. The code responsible for this is in update_task_states.rb

Reply:
now I got it, done